### PR TITLE
Add OOP principles

### PR DESCRIPTION
**Pull request by NatashaBel**
photo_2024-10-06 17 39 02

### CODE REVIEW

File: src/main/java/entity/AbstractEntity.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package entity;
+
+import java.util.UUID;
+
+public abstract class AbstractEntity {
```
Comment (AtticSiol) [inline src/main/java/entity/AbstractEntity.java]: rename it more specifically

File: src/main/java/model/ticket/Ticket.java [hunk 2]
```diff
@@ -103,6 +96,15 @@ private void setEventCode(short eventCode) {
...
+    public void print() {
+        System.out.println("Class content: " + this);
+    }
+
+    public void share(){
+        System.out.println("Default share");
+    }
+
     @Override
     public String toString() {
```
Comment (AtticSiol) [inline src/main/java/model/ticket/Ticket.java:124]: after equals and hashCode

File: src/main/java/model/ticket/Ticket.java [hunk 3]
```diff
@@ -103,6 +96,15 @@ private void setEventCode(short eventCode) {
         this.eventCode = eventCode;
     }
 
+    @Override
+    public void print() {
```
Comment (AtticSiol) [inline src/main/java/model/ticket/Ticket.java]: move in the end

File: src/main/java/model/ticket/Ticket.java [hunk 4]
```diff
@@ -103,6 +96,15 @@ private void setEventCode(short eventCode) {
...
+        System.out.println("Class content: " + this);
+    }
+
+    public void share(){
```
Comment (AtticSiol) [inline src/main/java/model/ticket/Ticket.java]: public void share()
 public void share(arg)
 public void share(arg, arg)

File: src/main/java/service/TicketService.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        //which is "print content in console".
+        Printable defaultPrint = new TicketService();
+        defaultPrint.print();
+        ticket1.print();
+        Printable fromClient = new Client(new Ticket());
+        fromClient.print();
+        //    @Override
+        //    public void print() {
+        //        System.out.println("Class content:" + " " + toString());
+        //    }
```
Comment (AtticSiol) [inline src/main/java/service/TicketService.java]: remove all redundant comments

File: src/main/java/entity/AbstractEntity.java (hunk 1 above)
Comment (NatashaBel) [reply src/main/java/entity/AbstractEntity.java]: Renamed Abstract class to BaseEntity

File: src/main/java/model/ticket/Ticket.java (hunk 2 above)
Comment (NatashaBel) [reply src/main/java/model/ticket/Ticket.java:124]: Moved method toString() after equals() and hashCode()

File: src/main/java/model/ticket/Ticket.java (hunk 3 above)
Comment (NatashaBel) [reply src/main/java/model/ticket/Ticket.java]: Moved the method print() to the end

File: src/main/java/model/ticket/Ticket.java (hunk 4 above)
Comment (NatashaBel) [reply src/main/java/model/ticket/Ticket.java]: Created new Interface Shareable with three methods.
Implemented these methods in Ticket class.

File: src/main/java/service/TicketService.java (hunk 5 above)
Comment (NatashaBel) [reply src/main/java/service/TicketService.java]: Removed all redundant comments

File: src/main/java/model/ticket/Ticket.java [hunk 6]
```diff
@@ -115,4 +132,25 @@ public String toString() {
...
+        System.out.println("Default share");
+    }
+
+    public void share(String phone) {
+        System.out.println("Default share by phone");
+    }
+
+    public void share(String phone, String email) {
+        System.out.println("Default share by phone and email");
+    }
```
Comment (AtticSiol) [inline src/main/java/model/ticket/Ticket.java]: good, just add @Override for each method

Comment (NatashaBel) [reply src/main/java/model/ticket/Ticket.java]: Added missed annotations

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**